fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryManager.java

```diff
@@ -101,17 +109,17 @@ public CompletableFuture<Void> mergeIndex(LucenePartitioner partitioner, LuceneA
         final Subspace subspace = state.indexSubspace;
         final KeyExpression rootExpression = state.index.getRootExpression();
 
+        final AgilityContext agilityContext = getAgilityContext(true);
         if (! (rootExpression instanceof GroupingKeyExpression)) {
             // TO DO: https://github.com/FoundationDB/fdb-record-layer/issues/2414
-            mergeIndex(analyzerWrapper, TupleHelpers.EMPTY, partitioner.isPartitioningEnabled() ? 0 : null);
-            return AsyncUtil.DONE;
+            return mergeIndex(analyzerWrapper, TupleHelpers.EMPTY, partitioner, agilityContext);
         }
         GroupingKeyExpression expression = (GroupingKeyExpression) rootExpression;
         final int groupingCount = expression.getGroupingCount();
 
         final RecordCursor<Tuple> cursor = new ChainedCursor<>(
                 state.context,
-                lastKey -> nextTuple(state.context, subspace, keyRange, lastKey, scanProperties, groupingCount),
+                lastKey -> agilityContext.apply(context -> nextTuple(context, subspace, keyRange, lastKey, scanProperties, groupingCount)),
```

Comment:
If the merge of an individual group takes more than 5 seconds the chained cursor won't be able to get the next group, and will fail. This has it use the AgilityContext to get the next group.